pkg/controller/v1beta1/inferenceservice/components/transformer.go

```diff
@@ -79,6 +88,29 @@ func (p *Transformer) Reconcile(isvc *v1beta1.InferenceService) error {
 		}),
 		Annotations: annotations,
 	}
+
+	// Need to wait for predictor URL in modelmesh deployment mode
+	if isvcutils.GetDeploymentMode(annotations, deployConfig) == constants.ModelMeshDeployment {
+		// check if predictor URL is populated
+		predictorURL := (*url.URL)(isvc.Status.Components["predictor"].URL)
+		if predictorURL == nil {
+			// transformer reconcile will retry every 3 second until predictor URL is populated
+			p.Log.Info("Transformer reconciliation is waiting for predictor URL to be populated")
+			return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
+		}
+
+		// add predictor host and protocol to metadata
+		isvc.ObjectMeta.Annotations["predictor-host"] = predictorURL.Host
```

Comment:
@chinhuang007 Can you help move this to a constant and rename following the internal annotation name convention
https://github.com/kserve/kserve/blob/master/pkg/constants/constants.go#L80